ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetTransactionReceipt.java

```diff
@@ -106,7 +107,23 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
               Bytes.fromBase64String(new String(receiveResponse.getPayload(), UTF_8)), false);
 
       privateTransaction = PrivateTransaction.readFrom(input);
-      privacyGroupId = receiveResponse.getPrivacyGroupId();
+      if (privateTransaction.getPrivacyGroupId().isPresent()) {
```

Comment:
yes it would make this piece cleaner - but you would have to build logic to ensure the database is mutually exclusive with respect to onchain/orion privacy groups which will be worse than this.